umbraco/Umbraco-CMS
# Anonymous telemetry

## Feature
For new installs and for upgrades this will add a new telemetry marker file at `/umbraco/telemetrics-id.umb` with a unique GUID to identify the site.

Then this is pinged home every 24 hours to send us a request. The use of this data is to help track the active number of installs in a more accurate way.

## Test Notes
* Verify marker file is created for you in a clean install and also an upgrade with a valid GUID as its contents
* Attach debugger & boot site
* Note with the debugger attached it will send the telemetry request to the DEV RainbowsRocks service
* Verify the request is sent to the service and get back a success status code.

## Things to consider
* Removing the marker file will stop the service from calling home
* Changing the contents of the marker file to be a non-valid GUID will stop it from calling from home
* If a user has decided to delete the marker as wanting to opt-out then the next time they upgrade the file will be re-added back
